Library/Homebrew/cmd/update.sh

```diff
@@ -218,8 +218,8 @@ merge_or_rebase() {
   if [[ "$DIR" = "$HOMEBREW_REPOSITORY" && -n "$HOMEBREW_UPDATE_TO_TAG" ]]
   then
     UPSTREAM_TAG="$(git tag --list |
-                    sort --field-separator=. --key=1,1nr -k 2,2nr -k 3,3nr |
-                    grep --max-count=1 '^[0-9]*\.[0-9]*\.[0-9]*$')"
+                    sort --field-separator=. -k 1,1nr -k 2,2nr -k 3,3nr -k 4,4nr |
+                    grep -E --max-count=1 '^[0-9]+\.[0-9]+\.[0-9]+(\.[0-9]+)?$')"
```

Comment:
right now the code depends on this behavior of `sort`. If it changes in the future OR is different on some Linux platform then it will not work as intended.  But, I guess, that's good enough™ :)